### ISSUE TITLE
Bug: 10 zones in my rainbird, only 7 in HomeKit

### ISSUE THREAD

**Issue by tmenguy** [homebridge-plugins/homebridge-rainbird#510]
### Describe The Bug

Great plugin, thx! ....But I have 10 Zones, and the plugin maps only 7 

### To Reproduce

well it is my setup :)
I have in Rainbird zone 1,2,3,4,5,6,7 an then it jumps to 11,12,13 , perhaps an issue because of that in the discovery code?

### Expected behavior

We should have 10 zones visible

### Relevant log output

```shell
[09/12/2023, 19:16:31] [RainBird] Model: ESP-Me, [Version: 2.9, Serial Number: 4231e59821e100, Zones: [1,2,3,4,5,6,7]]
[09/12/2023, 19:16:31] [RainBird] Restoring existing accessory from cache: ESP-Me
[09/12/2023, 19:16:31] [RainBird] IrrigationSystem ESP-Me Config: {"ipaddress":"XXX.XX.XXX.XXX","password":"XXXXXXXX","showValveSensor":true,"showProgramASwitch":true,"showProgramBSwitch":true,"showProgramCSwitch":true,"showStopIrrigationSwitch":true,"showZoneValve":true,"showDelayIrrigationSwitch":true,"syncTime":false,"showRainSensor":false,"showProgramDSwitch":false,"includeZones":"","irrigationDelay":1,"showRequestResponse":false,"minValueRemainingDuration":0,"maxValueRemainingDuration":3600}
[09/12/2023, 19:16:31] [RainBird] Restoring existing accessory from cache: Zone 1
[09/12/2023, 19:16:31] [RainBird] Restoring existing accessory from cache: ESP-Me-1
[09/12/2023, 19:16:31] [RainBird] Restoring existing accessory from cache: Zone 2
[09/12/2023, 19:16:31] [RainBird] Restoring existing accessory from cache: ESP-Me-2
[09/12/2023, 19:16:31] [RainBird] Restoring existing accessory from cache: Zone 3
[09/12/2023, 19:16:31] [RainBird] Restoring existing accessory from cache: ESP-Me-3
[09/12/2023, 19:16:31] [RainBird] Restoring existing accessory from cache: Zone 4
[09/12/2023, 19:16:31] [RainBird] Restoring existing accessory from cache: ESP-Me-4
[09/12/2023, 19:16:32] [RainBird] Restoring existing accessory from cache: Zone 5
[09/12/2023, 19:16:32] [RainBird] Restoring existing accessory from cache: ESP-Me-5
[09/12/2023, 19:16:32] [RainBird] Restoring existing accessory from cache: Zone 6
[09/12/2023, 19:16:32] [RainBird] Restoring existing accessory from cache: ESP-Me-6
[09/12/2023, 19:16:32] [RainBird] Restoring existing accessory from cache: Zone 7
[09/12/2023, 19:16:32] [RainBird] Restoring existing accessory from cache: ESP-Me-7
[09/12/2023, 19:16:32] [RainBird] Restoring existing accessory from cache: ESP-Me-pgm-A
[09/12/2023, 19:16:32] [RainBird] Restoring existing accessory from cache: ESP-Me-pgm-B
[09/12/2023, 19:16:32] [RainBird] Restoring existing accessory from cache: ESP-Me-pgm-C
[09/12/2023, 19:16:32] [RainBird] Restoring existing accessory from cache: ESP-Me-stop
[09/12/2023, 19:16:32] [RainBird] Restoring existing accessory from cache: ESP-Me-delay
```

### Config for homebridge-rainbird

```shell
{
            "name": "RainBird",
            "devices": [
                {
                    "ipaddress": "10.0.XXX.XXX",
                    "password": "XXXXX",
                    "showValveSensor": true,
                    "showProgramASwitch": true,
                    "showProgramBSwitch": true,
                    "showProgramCSwitch": true,
                    "showStopIrrigationSwitch": true,
                    "showZoneValve": true,
                    "showDelayIrrigationSwitch": true,
                    "syncTime": false
                }
            ],
            "platform": "RainBird"
        }
```

### Device & Model

ESPE ME

### Node.js Version

v18.16.0

### NPM Version

??

### Homebridge Version

v1.7.0

### Homebridge Rainbird Plugin Version

v1.11.2

### Homebridge Config UI X Plugin Version

v4.54.1

### Operating System

Raspbian GNU/Linux Bullseye (11)

**Comment by mantorok1** [comment]
Hi @tmenguy,
I've not seen a Rainbird which has a gap in the zones. I'll need to get some additional logs from you to help sort this out for you. Firstly could you turn on "Show RainBird requests and responses in the log" and restart the plugin/homebridge.
When it restarts the logs should now show the requests it makes to the RainBird controller and the responses it gets back. For now the one I'm looking for should look something like this:
```
Response: [131,0,...] [AvailableZonesResponse]
```
Could you please copy this and paste it into the comments. This is the only log message I need so you don't need to paste all the others.

Please note that your model is not fully supported yet by the plugin and you may receive a message in the logs about it. If you would like me to add better support can you please paste the details requested from those logs. The log message will begin like this:
```
This plugin does not fully support your RainBird model ...
```

**Comment by tmenguy** [comment]
Hi,

for the begging of the log;

[RainBird] [10.0.4.85] Request:  [2] [ModelAndVersionRequest]
[09/12/2023, 20:34:23] Homebridge v1.7.0 (HAP v0.11.1) (Homebridge D85A) is running on port 51818.
[09/12/2023, 20:34:23] [RainBird] [10.0.4.85] Response: [130,0,7,2,9] [ModelAndVersionResponse]
[09/12/2023, 20:34:24] [RainBird] [10.0.4.85] Request:  [5] [SerialNumberRequest]
[09/12/2023, 20:34:24] [RainBird] [10.0.4.85] Response: [133,66,49,229,152,33,225,0,0] [SerialNumberResponse]
[09/12/2023, 20:34:24] [RainBird] [10.0.4.85] Request:  [3,0] [AvailableZonesRequest]
[09/12/2023, 20:34:25] [RainBird] [10.0.4.85] Response: [131,0,127,28,0,0] [AvailableZonesResponse]
[09/12/2023, 20:34:25] [RainBird] [10.0.4.85] Request:  [72] [IrrigationStateRequest]
[09/12/2023, 20:34:26] [RainBird] [10.0.4.85] Response: [200,1] [IrrigationStateResponse]
[09/12/2023, 20:34:26] [RainBird] [10.0.4.85] Request:  [59,0] [ProgramZoneStateRequest]
[09/12/2023, 20:34:26] [RainBird] [10.0.4.85] Response: [187,0,0,0,0,0,0,0,0,255,0,0] [ProgramZoneStateResponse]
[09/12/2023, 20:34:26] [RainBird] [10.0.4.85] Request:  [62] [RainSensorStateRequest]
[09/12/2023, 20:34:26] [RainBird] [10.0.4.85] Response: [190,0] [RainSensorStateResponse]
[09/12/2023, 20:34:26] [RainBird] [DEBUG] {"modelNumber":7,"model":"ESP-Me","version":"2.9","serialNumber":"4231e59821e100","zones":[1,2,3,4,5,6,7]}
[09/12/2023, 20:34:26] [RainBird] Model: ESP-Me, [Version: 2.9, Serial Number: 4231e59821e100, Zones: [1,2,3,4,5,6,7]]
[09/12/2023, 20:34:26] [RainBird] Restoring existing accessory from cache: ESP-Me
[09/12/2023, 20:34:26] [RainBird] [DEBUG] IrrigationSystem: ESP-Me Using Device Config Logging: debug
[09/12/2023, 20:34:26] [RainBird] [DEBUG] IrrigationSystem: ESP-Me Using Platform Config refreshRate: 300

There is no "This plugin does not fully support your RainBird model" in the logs ...

Thanks a lot for your help (and plugin :))

BR
Thomas

**Comment by mantorok1** [comment]
Hi @tmenguy, thanks for that. I think I've identified the issue. There is a new beta version (1.11.3 beta 2) which should fix it up. Try it out and let me know if it solves the problem.

The "This plugin does not fully support your RainBird model" log message will only show when a program/zone is running. Try starting a program and then see if the log message is displayed.

**Comment by tmenguy** [comment]
Cool excellent Will try asap.
Is there a way to name the zones (in HomeKit) ? Either by getting the ones from the rainbird directly, or if not possible in the config?

**Comment by mantorok1** [comment]
Unfortunately I'm not aware of any way to retrieve the zone names. I suspect that the zone names are only stored on the app and not in the RainBird controller. However you can set them within the Home app on your iPhone, just go into the config for the Irrigation System or Valve accessory and change the name.

**Comment by tmenguy** [comment]
Ok tried the Beta ... works now! I see all my zones, thx a lot!

**Comment by tmenguy** [comment]
The hierarchy in Home kit is just strange : I have a "irrigations zones" a top level with inside the zones 11,12,13 and another irrigation zones with inside the zones 1,2,3,4,5,6,7. So at high level in HomeKit UI it says "4 zones Off" and not 11

**Comment by mantorok1** [comment]
Could you please supply some screenshots of what you are seeing

**Comment by tmenguy** [comment]
Well a bit strange as it is on my Mac and different than on my phone, 

here is the Mac:
<img width="541" alt="image" src="https://github.com/donavanbecker/homebridge-rainbird/assets/2277897/7c981302-954e-4da6-9d9b-f815ae2f3ad2">

<img width="541" alt="image" src="https://github.com/donavanbecker/homebridge-rainbird/assets/2277897/4d335ab4-9fd4-4470-8646-d404055320cb">

as you can see 4 Zones at high level the other 7 are in the leveller below

**Comment by tmenguy** [comment]
and on the phones everything seems as expected : only one level in the settings

**Comment by mantorok1** [comment]
Interesting. I suspect this may be an issue with MacOS's Home app given that it's working fine in iOS. It wouldn't be the first time!

It may be a side effect of zones 11,12 & 13 being added later by the new version of the plugin. One thing you could try is uninstalling the plugin and make sure its been removed from the Home app (macOS & iOS). Then install it again from scratch

**Comment by tmenguy** [comment]
anyway, to me the current bug can be closed and will be fixed in the next release :) Thanks again.
You want me to close now, or you prefer to close when you release the beta as official?
BR
Thomas

**Comment by mantorok1** [comment]
Hi @tmenguy, feel free to close this issue

If you don't mind could you please let me know if you're seeing the time remaining when a zone is running (both in the Home app and in the logs). Also when a program is running is the switch in the Home app in the On state for the duration of the program?

**Comment by donavanbecker** [comment]
Fixed with `v1.11.3`
